web/src/components/storage/DASDTable.tsx

```diff
@@ -144,36 +147,40 @@ const Actions = ({ devices, isDisabled }) => {
   );
 };
 
-const filterDevices = (devices, from, to) => {
-  const allChannels = devices.map((d) => d.hexId);
+const filterDevices = (devices: DASDDevice[], from: string, to: string): DASDDevice[] => {
+  const allChannels: number[] = devices.map((d) => d.hexId);
   const min = hex(from) || Math.min(...allChannels);
   const max = hex(to) || Math.max(...allChannels);
 
   return devices.filter((d) => d.hexId >= min && d.hexId <= max);
 };
 
-export default function DASDTable({ state, dispatch }) {
+export default function DASDTable() {
+  const devices = useDASDDevices();
+  const { mutate: changeFilter } = useFilterDASDChange();
+  const { mutate: changeSelected } = useSelectedDASDChange();
+  const { minChannel, maxChannel } = useFilterDASD();
+  const selectedDevices = useSelectedDASD();
+
   const [sortingColumn, setSortingColumn] = useState(columns[0]);
-  const [sortDirection, setSortDirection] = useState("asc");
+  const [sortDirection, setSortDirection]: ["asc" | "desc", Dispatch<SetStateAction<"asc" | "desc">>] = useState("asc");
 
   const sortColumnIndex = () => columns.findIndex((c) => c.id === sortingColumn.id);
-  const filteredDevices = filterDevices(state.devices, state.minChannel, state.maxChannel);
-  const selectedDevicesIds = state.selectedDevices.map((d) => d.id);
+  const filteredDevices = filterDevices(devices, minChannel, maxChannel);
+  const selectedDevicesIds = selectedDevices.map((d) => d.id);
 
   // Selecting
   const selectAll = (isSelecting = true) => {
-    const type = isSelecting ? "SELECT_ALL_DEVICES" : "UNSELECT_ALL_DEVICES";
-    dispatch({ type, payload: { devices: filteredDevices } });
+    changeSelected({ unselect: !isSelecting, devices: filteredDevices })
```

Comment:
This is something we should talk about. If we actually desired this kind of behavior and the naming for these queries.

To keep such a behavior, I think we should bear in mind

* consistency, all the pages should behave in an similar way on that regard.
* usability, UI should provide a clear, usable way to both, let users know that there are filters applied and an easy to find reset / clear action
* naming, kind of DX, look for a naming strategy for these queries in order to quickly distinguish them from the API queries.

But please, go ahead with this change as it is. Just open, create a follow-up card (if needed)

Reply:
Additionally, we might consider to use a single query for all the DASD filters.

Reply:
> Additionally, we might consider using a single query for all the DASD filters.

Good idea, something to evaluate in next iteration I guess.

Reply:
> This is something we should talk about. If we actually desired this kind of behavior and the naming for these queries.
> 
> To keep such a behavior, I think we should bear in mind
> 
> * consistency, all the pages should behave in an similar way on that regard.
> * usability, UI should provide a clear, usable way to both, let users know that there are filters applied and an easy to find reset / clear action
> * naming, kind of DX, look for a naming strategy for these queries in order to quickly distinguish them from the API queries.
> 
> But please, go ahead with this change as it is. Just create a follow-up card (if needed)

Thinking about this twice, I believe now that would be better to

* Do not keep filters applied after navigating. It's kind of unexpected and it will not play well for all pages
* Rework this part before merging the PR. Don't worry, I can help here.

You may argue that it's a complex page and the user could benefit of landing in the page again with the previous filtering already applied. But we shouldn't compromise consistency among pages because a special use case. Instead, we should arrange time to work on improving the DASD page filters for offering easy to use yet powerful filtering options to _DASD users_.

So, I'll try to get rid of query for filtering here and use a stateful page approach instead.